cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -84,6 +84,373 @@ constexpr inline auto is_supported_construction_value_type()
 // Helper functions for `fixed_point` type
 namespace detail {
 
+/**
+ * @brief Helper struct containing large compile-time-computed powers of 10
+ */
+struct PowersOf10 {
```

Comment:
Suggestion: I think you can replace this large boilerplate struct with a simple recursive function template: 

```
template <int exp>
constexpr __uint128_t power_of_10()
{
    static_assert(exp >= 0);
    if constexpr (exp == 0) return __uint128_t(1ULL);
    else return power_of_10<exp-1>() * __uint128_t(10ULL);
}
```

You could add some template smarts to automatically determine the type based on the size of `exp` if you want.

https://godbolt.org/z/fMvYEbY4n